### PR TITLE
 obsloctap skeletal

### PR DESCRIPTION
**Pull request by womullan**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,22 +11,8 @@ Phalanx is developed by the `Vera C. Rubin Observatory <https://www.lsst.org/>`_
...
-* `data-int.lsst.cloud <https://data-int.lsst.cloud/argo-cd>`__ (idfint)
-* `data.lsst.cloud <https://data.lsst.cloud/argo-cd>`__ (idfprod)
-
-Telescope and Site:
-
-* `tucson-teststand.lsst.codes <https://tucson-teststand.lsst.codes/argo-cd>`__ (tucson-teststand)
-* `base-lsp.lsst.codes <https://base-lsp.lsst.codes/argo-cd>`__ (base)
-* `summit-lsp.lsst.codes <https://summit-lsp.lsst.codes/argo-cd>`__ (summit)
+The environments managed by Argo CD using configuration in this repository are detaild
+on `phalanx.lsst.io <https://phalanx.lsst.io/environments/index.html>`
```
Comment (rra) [inline README.rst:15]:
```suggestion
The environments managed by Argo CD using configuration in this repository are detailed
on `phalanx.lsst.io <https://phalanx.lsst.io/environments/>`__.
```

File: applications/obsloctap/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          volumeMounts:
+            - name: "butler-secrets"
+              mountPath: "/home/worker/.lsst/"
+            - name: "butler-secrets-raw"
+              mountPath: "/home/worker/secrets-raw/"
+              readOnly: true
+      containers:
+      - name: obsloctap
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion}}
+        imagePullPolicy: Always
```
Comment (rra) [inline applications/obsloctap/templates/deployment.yaml:56]: As above.
```suggestion
        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
```

File: applications/obsloctap/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        volumeMounts:
+          - name: butler-secrets
+            mountPath: "/home/worker/.lsst/"
+          {{- with .Values.config.volume_mounts }}
+          {{- . | toYaml | nindent 10 }}
+          {{- end }}
+        command:
+          - /bin/bash
+          - -c
+          - panel serve dashboard_gen3.py --port 8080 --prefix /obsloctap --allow-websocket-origin {{ .Values.global.host }} --static-dirs assets=./assets
```
Comment (rra) [inline applications/obsloctap/templates/deployment.yaml:71]: When using the FastAPI app template, the default command will be fine (and this command isn't the correct one).

```suggestion
```

File: applications/obsloctap/templates/ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+kind: GafaelfawrIngress
+metadata:
+  name: "obsloctap"
+  labels:
+    {{- include "obsloctap.labels" . | nindent 4 }}
+config:
+  baseUrl: {{ .Values.global.baseUrl | quote }}
+  scopes:
+    all:
+      - "exec:portal"
```
Comment (rra) [inline applications/obsloctap/templates/ingress.yaml:11]:
```suggestion
      - "read:tap"
```

File: applications/obsloctap/templates/ingress.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  loginRedirect: true
+  delegate:
+    internal:
+      scopes: []
+      service: "obsloctap"
```
Comment (rra) [inline applications/obsloctap/templates/ingress.yaml:16]: You probably will not need any of this for an API service.
```suggestion
```

**Review by rra — COMMENTED** [review]
Also add a file `obsloctap-application.yaml` in `environments/templates` that creates the Argo CD application, and add `obsloctap` to `environments/values.yaml` disabled by default. See step 3 at https://phalanx.lsst.io/developers/add-application.html

File: applications/obsloctap/templates/ingress.yaml (hunk 4 above)
Comment (womullan) [reply applications/obsloctap/templates/ingress.yaml:11]: should this be read:all or read:any

Comment (womullan) [reply applications/obsloctap/templates/ingress.yaml:11]: I do probably want a web page as well as an API

**Comment by womullan** [comment]
application added - and other changes accepted or made

**Comment by womullan** [comment]
thanks for all the pointers

**Pull request merged**